src/PrestaShopBundle/Controller/Admin/CommonController.php

```diff
@@ -357,15 +359,30 @@ public function searchGridAction(
         $redirectRoute,
         array $redirectQueryParamsToKeep = []
     ) {
-        /** @var ResponseBuilder $responseBuilder */
-        $responseBuilder = $this->get('prestashop.bundle.grid.response_builder');
         /** @var GridDefinitionFactoryInterface $definitionFactory */
         $definitionFactory = $this->get($gridDefinitionFactoryServiceId);
+        $reflectionClass = new ReflectionClass($definitionFactory);
```

Comment:
I have always been told to never use Reflection in production 😮(ping @PierreRambaud any opinion on this ?)

Reply:
I'm not a big fan either but I only see two alternatives:
- a try/catch block witch would catch a very generic `FatalThrowaleError` (if ti works)
- adding an interface to all grid definitions which manage the GRID_ID which would allow to use `instanceof`

Reply:
I ❤️ the interface solution

Reply:
`Interface` FTW

Reply:
The only issue about the interface solution is that it brings more complexity for module developers, we will have to update the doc accordingly

And I'm afraid it could create a BC break 🤔 as I'm not sure exactly what the interface should contain:
`public function getFilterId(): string` maybe?

Reply:
This complexity already exist in PrestaShop, and if you want to help developers feel free to add an abstract class in addition of the Interface.

Reply:
No BC break I think 🤔
- create new Interface `IdentifiedGridDefinitionFactoryInterface` that extends existing `GridDefinitionFactoryInterface`
- apply to all existing grids in Core
- check if given definition is instance of `IdentifiedGridDefinitionFactoryInterface` to use the ID